src/main/scala/org/broadinstitute/hail/sparkextras/AdjustedPartitionsRDD.scala

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.hail.sparkextras
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{Dependency, NarrowDependency, Partition, TaskContext}
+
+import scala.reflect.ClassTag
+
+case class AdjustedPartitionsRDDPartition[T](index: Int, parent: Partition, adj: Adjustment[T]) extends Partition
+
+case class Adjustment[T](originalIndex: Int,
+  drop: Option[Iterator[T] => Iterator[T]],
+  take: Seq[(Int, Iterator[T] => Iterator[T])])
```

Comment:
Make this more general: just have a list of partitions to take from with filters.